### DISCUSSION TITLE
Infinite Loop in Deformable World Simulation Step

### DISCUSSION THREAD

**Discussion by edwin-pan** [bulletphysics/bullet3#4276] [Q&A]
I had been using an older version of Bullet for simulating deformable objects (works quite well). Today, while switching to the latest hotness, I now run into an infinite loop in the `buildTreeBottomUp(...)` function in `btSoftBody.cpp`. Specifically, I moved from [this version](https://github.com/bulletphysics/bullet3/blob/830f0a9565b1829a07e21e2f16be2aa9966bd28c/src/BulletSoftBody/btSoftBody.cpp) of `btSoftBody.cpp` to [this version](https://github.com/bulletphysics/bullet3/blob/master/src/BulletSoftBody/btSoftBody.cpp). 

In the first pass call to `buildTreeBottomUp(...)` from [here](https://github.com/bulletphysics/bullet3/blob/19aa740d8464cd5984f3fd37f285a610dedd4163/src/BulletSoftBody/btSoftBody.cpp#L2688), the tree build has values of `N`:
`N: 400,N: 226,N: 133,N: 84,N: 57,N: 43,N: 35,N: 31,N: 29,N: 28,N: 27,N: 26,N: 25,N: 24,N: 23,N: 22,N: 21,N: 20,N: 19,N: 18,N: 17,N: 16,N: 15,N: 14,N: 13,N: 12,N: 11,N: 10,N: 9,N: 8,N: 7,N: 6,N: 5,N: 4,N: 3,N: 2`
and then continues onward.

The next call to `buildTreeBottomUp(...)` happens in the [rebuildNodeTree(...)](https://github.com/bulletphysics/bullet3/blob/19aa740d8464cd5984f3fd37f285a610dedd4163/src/BulletSoftBody/btSoftBody.cpp#L2693) call  and can be found [here](https://github.com/bulletphysics/bullet3/blob/19aa740d8464cd5984f3fd37f285a610dedd4163/src/BulletSoftBody/btSoftBody.cpp#L2729), and the tree rebuild gets stuck at `N=17`:
`N: 274,N: 148,N: 84,N: 52,N: 37,N: 29,N: 25,N: 23,N: 22,N: 21,N: 20,N: 19,N: 18,N: 17,N: 17,N: 17,N: 17,N: 17, .............`

What is the purpose of this tree rebuilding process and how might I go about debugging this?

**Comment by 36yang** [comment]
I also encountered the same problem
